### PR TITLE
rewrite abspath(__file__) to __file__ in py39+

### LINKED ISSUE #576: py39+: abspath(__file__) -> __file__
since `__file__` is now an absolute path in python3.9+

### PR DESCRIPTION
**Pull request by MarcoGorelli**
resolves #576

### CODE REVIEW

File: tests/features/abspath_file_test.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        pytest.param(
+            'import os\n'
+            'os.path.abspath(__file__)',
+            'import os\n'
+            '__file__',
+            id='os.path.abspath',
+        ),
+    ),
+)
+def test_fix_encode(s, expected):
```
Comment (asottile) [inline tests/features/abspath_file_test.py:26]: heh -- these got copy pasted

Comment (MarcoGorelli) [reply tests/features/abspath_file_test.py]: caught me 😳

**Review by asottile — APPROVED** [review]

**Comment by adamantike** [comment]
I will try to create a simple example and open a separate issue, but this change is breaking a script we are running with Python 3.9 and `alembic`.

We run `alembic upgrade --sql head`, while having a line like this in `alembic/env.py`:
```python
sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
```

That works, but after running `pyupgrade`, that line gets replaced with:
```python
sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))
```

And the `alembic` command fails. Printing `__file__` shows:
```python
ipdb> __file__
'alembic/env.py'
ipdb> import sys; sys.version
'3.9.13 (main, Aug  2 2022, 11:20:39) \n[GCC 10.2.1 20210110]'
```

**Pull request merged**